internal/pkg/cli/svc_deploy.go

```diff
@@ -221,21 +218,29 @@ func (o *deploySvcOpts) Execute() error {
 			return err
 		}
 	}
+	raw, err := o.ws.ReadWorkloadManifest(o.name)
+	if err != nil {
+		return fmt.Errorf("read manifest file for %s: %w", o.name, err)
+	}
+	interpolated, err := o.newInterpolator(o.appName, o.envName).Interpolate(string(raw))
+	if err != nil {
+		return fmt.Errorf("interpolate environment variables for %s manifest: %w", o.name, err)
+	}
 	mft, err := workloadManifest(&workloadManifestInput{
-		name:         o.name,
-		appName:      o.appName,
-		envName:      o.envName,
-		interpolator: o.newInterpolator(o.appName, o.envName),
-		ws:           o.ws,
-		unmarshal:    o.unmarshal,
-		sess:         o.envSess,
+		name:            o.name,
+		appName:         o.appName,
+		envName:         o.envName,
+		interpolatedMft: interpolated,
+		unmarshal:       o.unmarshal,
+		sess:            o.envSess,
 	})
 	if err != nil {
 		return err
 	}
 	if o.forceNewUpdate && o.svcType == manifestinfo.StaticSiteType {
 		return fmt.Errorf("--%s is not supported for service type %q", forceFlag, manifestinfo.StaticSiteType)
 	}
+	o.rawMft = interpolated
```

Comment:
can we move this line up to right after we get the value for `interpolated`? or move it inside of `workloadManifest` so that we don't have to move the interpolation logic out.